src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginTest.groovy

```diff
@@ -228,33 +236,56 @@ class ScalafixPluginTest extends Specification {
+        task.mode == ScalafixMainMode.IN_PLACE
     }
 
-//    def 'scalafix uses the .scalafix config file from the subproject by default'() {
-//        setup:
-//        Project subproject = ProjectBuilder.builder().withName('the-subproject')
-//                .withParent(project).build()
-//
-//        when:
-//        applyScalafixPlugin(subproject)
-//
-//        then:
-//        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
-//        task.configFile.get().asFile.path == "${subproject.projectDir}/.scalafix.conf"
-//    }
-//
-//    def 'scalafix uses the .scalafix config file from the root project as the file is not present in the subproject'() {
-//        setup:
-//        Project subproject = ProjectBuilder.builder().withName('the-subproject')
-//                .withParent(project).build()
-//
-//        when:
-//        applyScalafixPlugin(subproject)
-//
-//        then:
-//        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
-//        task.configFile.get().asFile.path == "${project.projectDir}/.scalafix.conf"
-//    }
+    def 'scalafix uses the .scalafix config file provided via extension'() {
+        setup:
+        Project subproject = ProjectBuilder.builder().withName('the-subproject')
+                .withParent(project).build()
+        subproject.projectDir.mkdir()
+        File subprojectScalafixConf = new File(subproject.projectDir, '.scalafix.conf')
+        subprojectScalafixConf.write 'rules = [Foo, Bar]'
+        File extensionScalafixConf = new File(subproject.projectDir, '.test-scalafix.conf')
+        extensionScalafixConf.write 'rules = [Foo, Bar]'
+
+        when:
+        applyScalafixPlugin(subproject, true, '', extensionScalafixConf)
+
+        then:
+        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
+        task.configFile.get().asFile.path == "${subproject.projectDir}/.test-scalafix.conf"
+    }
+
+    def 'scalafix uses the .scalafix config file from the subproject if it has not been provided via extension'() {
+        setup:
+        Project subproject = ProjectBuilder.builder().withName('the-subproject')
+                .withParent(project).build()
+        subproject.projectDir.mkdir()
+        File subprojectScalafixConf = new File(subproject.projectDir, '.scalafix.conf')
+        subprojectScalafixConf.write 'rules = [Foo, Bar]'
+
+        when:
+        applyScalafixPlugin(subproject, true, '', null)
+
+        then:
+        ScalafixTask task = subproject.tasks.getByName('checkScalafixMain')
+        task.configFile.get().asFile.path == "${subproject.projectDir}/.scalafix.conf"
+    }
+
+    def 'scalafix uses the .scalafix config file from the root project as the file is not present in the subproject and\
+ it is not specified in the extension'() {
+        setup:
+        Project subproject = ProjectBuilder.builder().withParent(project).withName('the-subproject').build()
+        scalafixConf.delete()
```

Comment:
I don't get why we need to delete the file here. Isn't it the file at the root project dir that is expected to be used?

Reply:
ok, I think I got it now. It has to do with https://github.com/cosmicsilence/gradle-scalafix/pull/9/files#r373798150 . I think the correct would be `null` (`<empty>`) as this file does not exist.